src/include/common/event_util.h

```diff
@@ -35,6 +35,14 @@ class EventUtil {
   }
 
  public:
+  /**
+   * EVENT_ACTIVATE_OR_TIMEOUT_ONLY can be supplied to event_new() to indicate that the event can only be activated
+   * by an explicit event_active or by a timeout firing.
+   */
+  static constexpr const int EVENT_ACTIVATE_OR_TIMEOUT_ONLY = -1;
```

Comment:
I think this actually forces static storage duration (i.e. prevents the compiler from completely eliding allocating it and just inlining the constant expression). I think you want to drop the `static` and then the `constexpr` is free to be elided as an allocated variable.

Reply:
I get `non-static data error cannot be constexpr: did you intend to make it static?` if I drop the static.

I wrote a quick test (attached) where the compiler at `O0` seems to be able to constant fold it away across multiple translation units, maybe it is fine?

![image](https://user-images.githubusercontent.com/8349071/94254948-7f099b00-fef5-11ea-9191-f57601bbe2a7.png)

Reply:
That's a class member though. `static` has different meanings at the namespace level, block level, and class member level.